solutions/src/router.js

```diff
@@ -0,0 +1,85 @@
+const http = require('http');
+const users = require('./static.js');
+const fs = require('fs');
+const pg = require('pg');
+const getData = require('./queries/getData.js');
+const postData = require('./queries/postData.js');
+const queryString = require('querystring');
+
+const router = (request, response) => {
+    const endpoint = request.url.split('/')[1];
+
+    if (endpoint === '') {
+        fs.readFile(__dirname + "/../public/index.html", function(error, file) {
+            if (error) {
+                response.writeHead(500, 'Content-Type:text/html');
+                response.end('<h1>Sorry, there was a problem loading the homepage</h1>');
+                console.log(error);
+            } else {
+                response.writeHead(200, {
+                    "Content-Type": "text/html"
+                });
+                response.end(file);
+            }
+        });
+    } else if (endpoint === "users") {
+        // TASK 1: replace the 3 lines below below with your own function that gets data from your database
+        getData((err, res) => {
+            if (err) {
+                response.writeHead(500, 'Content-Type:text/html');
+                response.end('<h1>Sorry, there was a problem getting the users</h1>');
+                console.log(error);
+            } else {
+                let output = JSON.stringify(res);
+                response.writeHead(200, {
+                    'content-type': 'application/json'
+                });
+                response.end(output);
+            }
+        });
+    } else if (endpoint === "create-user") {
+        let data = '';
+        request.on('data', function(chunk) {
+            data += chunk;
+        });
+        request.on('end', () => {
+            const name = queryString.parse(data).name;
+            const location = queryString.parse(data).location;
+            postData(name, location, (err, res) => {
+                if (err) {
+                    response.writeHead(500, 'Content-Type:text/html');
+                    response.end('<h1>Sorry, there was a problem adding that user</h1>');
+                    console.log(err)
+                }
+            });
+            response.writeHead(200, {
+                "Content-Type": "text/html"
+            });
+            fs.readFile(__dirname + "/../public/index.html", function(error, file) {
```

Comment:
`readFile` and `postData` are called on the same tick. That means that their callbacks might not execute in sequence. That is, you may be left in a position where you read `index.html` successfully and therefore end the response, but the `postData` callback hasn't fired yet. If there is then something wrong with the DB operation, you will try to call `response.end` again.

I would respond with the `index.html` only _after_ `postData` has successfully completed.